File: tui/src/scala/tui/Viewport.scala
package tui

case class Viewport(
    var area: Rect,
    resizeBehavior: ResizeBehavior
)

object Viewport {
  // UNSTABLE
  def fixed(area: Rect): Viewport =
    Viewport(
      area,
      resizeBehavior = ResizeBehavior.Fixed
    )
}

File: tui/src/java/tui/Viewport.java
package tui;

public final class Viewport {
  public Rect area;
  public final ResizeBehavior resizeBehavior;

  public Viewport(Rect area, ResizeBehavior resizeBehavior) {
    this.area = area;
    this.resizeBehavior = resizeBehavior;
  }

  // UNSTABLE
  public static Viewport fixed(Rect area) {
    return new Viewport(area, ResizeBehavior.Fixed);
  }
}
